tests/unit/features/partitions/test_lifecycle_manager.py

```diff
@@ -0,0 +1,138 @@
+
+from string import ascii_lowercase
+from typing import Any, Dict
+
+import pytest
+from hypothesis import HealthCheck, given, settings, strategies
+
+from craft_parts import errors
+from craft_parts.lifecycle_manager import LifecycleManager
+
+
+class TestPartitionsSupport:
+    """Verify LifecycleManager supports partitions."""
+
+    @pytest.fixture
+    def parts_data(self) -> Dict[str, Any]:
+        return {"parts": {"foo": {"plugin": "nil"}}}
+
+    @pytest.mark.parametrize("partitions", [["default"], ["default", "kernel"]])
+    def test_project_info(self, new_dir, parts_data, partitions):
+        """Verify partitions are parsed and passed to ProjectInfo."""
+        lifecycle = LifecycleManager(
+            parts_data,
+            application_name="test_manager",
+            project_name="project",
+            cache_dir=new_dir,
+            work_dir="work_dir",
+            arch="aarch64",
+            parallel_build_count=16,
+            custom="foo",
+            partitions=partitions,
+        )
+        info = lifecycle.project_info
+
+        assert info.application_name == "test_manager"
+        assert info.project_name == "project"
+        assert info.target_arch == "arm64"
+        assert info.arch_triplet == "aarch64-linux-gnu"
+        assert info.parallel_build_count == 16
+        assert info.dirs.parts_dir == new_dir / "work_dir" / "parts"
+        assert info.dirs.stage_dir == new_dir / "work_dir" / "stage"
+        assert info.dirs.prime_dir == new_dir / "work_dir" / "prime"
+        assert info.custom_args == ["custom"]
+        assert info.custom == "foo"
+        assert info.partitions == partitions
+
+    @pytest.mark.parametrize("partitions", [None, []])
+    def test_no_partitions(self, new_dir, parts_data, partitions):
+        """Raise an error if the partitions feature is enabled but not defined."""
+        with pytest.raises(errors.FeatureEnabled) as raised:
+            LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+        assert (
+            raised.value.message
+            == "Partition feature is enabled but no partitions are defined."
+        )
+
+    @pytest.mark.parametrize(
+        "partitions", [["defaulta"], ["kernel"], ["kernel", "default"]]
+    )
+    def test_default_not_first(self, new_dir, parts_data, partitions):
+        """Raise an error if the first partition is not 'default'."""
+        with pytest.raises(ValueError) as raised:
+            LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+        assert str(raised.value) == "First partition must be 'default'."
+
+    @settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+    @given(
```

Comment:
Inspired by https://github.com/canonical/craft-application/pull/20/files#diff-8d3623d9812e6ea2183ad0ed6faa2bcf4ca22b81c744497e4340ef8affe67614 (cc @lengau for review of this test)

Reply:
Looks good to me! I've got my tweezers out for nitpicking, but it's purely informational, so feel free to go 
![](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExcjU5eHY2MnNvZDYybzBodXZ0Z3E0M3kzaXVueGt0bm1laTdjaGhhOCZlcD12MV9naWZzX3NlYXJjaCZjdD1n/z4hS2bLjDNVU4PXcnI/giphy.gif)

Reply:
Also as a side note: This is a fantastic use case for hypothesis!